pkg/endpoint/endpoint.go

```diff
@@ -2341,9 +2341,16 @@ func (e *Endpoint) identityLabelsChanged(owner Owner, myChangeRev int) error {
 	e.Mutex.Unlock()
 
 	if readyToRegenerate {
-		e.Regenerate(owner, "updated security labels")
+		status := e.Regenerate(owner, "updated security labels")
+		go func(endpointBuildStatus <-chan bool, e *Endpoint) {
+			buildStatus := <-endpointBuildStatus
+			if !buildStatus {
+				e.Mutex.Lock()
+				e.SetStateLocked(StateRegenerating, "Endpoint regeneration failed")
```

Comment:
I understand the issue you posit in the PR description, but is `StateRegenerating` the correct state transition? The endpoint isn't really regenerating at this point.

Reply:
Would allowing `StateWaitingToRegenerate` -> `StateWaitingToRegenerate` state changes work better?  I think that might queue up more builds than today since we call `e.Regenerate` if `readyToRegenerate` is true (https://github.com/cilium/cilium/blob/master/pkg/endpoint/endpoint.go#L2389). That might be ok, though, since those builds would be no-ops (I think? they would be rejected in the queue).

Reply:
I just think that requiring an endpoint to be in a specific state when we request to regenerate it is bad. It introduces more state checks, which are prone to error, and it's launched asynchronously anyway. While this patch may address a specific issue, it's glossing over a fundamental design flaw in how endpoints are managed. To me, the real fix is the following: we should queue up builds with some type of metadata associated with each build indicating what the change was, and aggregate things that are queued up per endpoint to reduce unneeded builds.